Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -99,11 +99,38 @@ void SHADER::SetProgramVariables()
   }
 }
 
+bool IsDualSourceBlendActive()
+{
+  if (!g_ActiveConfig.backend_info.bSupportsDualSourceBlend)
+    return false;
+
+  GLboolean blendEnabled;
+  glGetBooleanv(GL_BLEND, &blendEnabled);
+  if (!blendEnabled) return false;
+
+  GLint blendSrc;
+  GLint blendSrcAlpha;
+  GLint blendDest;
+  GLint blendDestAlpha;
+  glGetIntegerv(GL_BLEND_SRC_RGB, &blendSrc);
+  glGetIntegerv(GL_BLEND_DST_RGB, &blendDest);
+  glGetIntegerv(GL_BLEND_SRC_ALPHA, &blendSrcAlpha);
+  glGetIntegerv(GL_BLEND_DST_ALPHA, &blendDestAlpha);
```

Comment:
I'm pretty sure you can just check the state of `bpmem.blendmode.srcfactor` and `bpmem.blendmode.dstfactor` here instead of querying OpenGL for it.